django/utils/encoding.py

```diff
@@ -196,20 +193,41 @@ def iri_to_uri(iri):
     return quote(force_bytes(iri), safe=b"/#%[]=:;$&()+,!?*@'~")
 
 
+_decode_ranges = [[45, 46, 95, 126], range(65, 91), range(97, 123), range(128, 256)]
```

Comment:
Could you add some explanatory comments for this stuff? I'm not too familiar with the relevant RFCs, etc.